### PR TITLE
Run pyupgrade

### PR DESCRIPTION
**Pull request by aseyboldt**
pyupgrade upgrades the syntax to make use of more recent python features. In our case those are mostly set literals and format strings. (https://github.com/asottile/pyupgrade)
It is a rather large diff, but I think it makes the source much nicer to read in a lot of places.

### CODE REVIEW

File: pymc3/backends/sqlite.py [hunk 1]
```diff
@@ -339,8 +339,8 @@ def load(name, model=None):
     db.connect()
     varnames = _get_table_list(db.cursor)
     if len(varnames) == 0:
-        raise ValueError(('Can not get variable list for database'
-                          '`{}`'.format(name)))
+        raise ValueError('Can not get variable list for database'
```
Comment (ColCarroll) [inline pymc3/backends/sqlite.py:343]: Heh.

File: pymc3/backends/sqlite.py [hunk 2]
```diff
@@ -366,14 +366,14 @@ def _get_table_list(cursor):
 
 
 def _get_var_strs(cursor, varname):
-    cursor.execute('SELECT * FROM [{}]'.format(varname))
+    cursor.execute(f'SELECT * FROM [{varname}]')
```
Comment (ColCarroll) [inline pymc3/backends/sqlite.py:370]: boy howdy we should get rid of this backend

File: pymc3/distributions/simulator.py [hunk 3]
```diff
@@ -43,9 +43,9 @@ def __init__(
             Distance functions. Available options are "gaussian_kernel" (default), "wasserstein",
             "energy" or a user defined function that takes epsilon (a scalar), and the summary
             statistics of observed_data, and simulated_data as input.
-            ``gaussian_kernel`` :math: `\sum \left(-0.5  \left(\frac{xo - xs}{\epsilon}\right)^2\right)`
-            ``wasserstein`` :math: `\frac{1}{n} \sum{\left(\frac{|xo - xs|}{\epsilon}\right)}`
-            ``energy`` :math: `\sqrt{2} \sqrt{\frac{1}{n} \sum \left(\frac{|xo - xs|}{\epsilon}\right)^2}`
+            ``gaussian_kernel`` :math: `\\sum \\left(-0.5  \\left(\frac{xo - xs}{\\epsilon}\right)^2\right)`
```
Comment (ColCarroll) [inline pymc3/distributions/simulator.py]: well, this is wonderful

File: pymc3/step_methods/hmc/quadpotential.py [hunk 4]
```diff
@@ -584,7 +584,7 @@ def update(self, sample, grad, tune):
 
     def raise_ok(self, vmap):
         if self._chol_error is not None:
-            raise ValueError("{0}".format(self._chol_error))
+            raise ValueError(f"{self._chol_error}")
```
Comment (ColCarroll) [inline pymc3/step_methods/hmc/quadpotential.py]: heh.

**Review by ColCarroll — COMMENTED** [review]
I like this -- I took a brief look and it seems better than before. Is it possible to hold onto it, and rerun after #4076 gets merged? I think that would be a painful rebase, and if this is just a script, it might be easier for you to do.

File: pymc3/distributions/simulator.py (hunk 3 above)
Comment (aseyboldt) [reply pymc3/distributions/simulator.py]: should be a raw string then...

File: pymc3/backends/sqlite.py (hunk 2 above)
Comment (aseyboldt) [reply pymc3/backends/sqlite.py:370]: yes...

**Comment by aseyboldt** [comment]
Waiting till after #4076.
It is trivial to recreate this PR

**Comment by MarcoGorelli** [comment]
Can this also be added to the CI process to make sure that outdated syntax isn't used in the future?

EDIT
----
if #4088 is accepted it's probably easiest to add this to the `pre-commit-config.yaml` file

**Comment by AlexAndorra** [comment]
+1 to add this to the `pre-commit-config.yaml` file as pre-commit hook -- will make our lives easier and our code consistently better!

**Comment by AlexAndorra** [comment]
#4088 is now merged, so pyupgrade can be added as a pre-commit hook quite easily now 🎉

**Comment by twiecki** [comment]
#4076 is merged so we can revisit this.

**Review by canyon289 — APPROVED** [review]

**Comment by twiecki** [comment]
Thanks Dr Seyboldt!

**Pull request merged**